prometheus/handler.go

```diff
@@ -47,6 +48,16 @@ type Handler struct {
 
 	opcount uint64
 	metrics metricStore
+	filters []byte
+}
+
+// FilterLabels will remove the passed labels from the observation
+func (h *Handler) FilterLabels(labelNames []string) {
+	b := make([][]byte, len(labelNames))
+	for i := range labelNames {
+		b[i] = []byte(labelNames[i])
+	}
+	h.filters = bytes.Join(b, []byte{0x00})
```

Comment:
This flattens all the filtered labels into a single byte slice so we dont have to do an O(n^2) lookup when we filter labels out

Reply:
Would be nice to add this comment into the code, it would be useful for people reading the code later.